diff --git a/backends/backend-sdl/build.gradle b/backends/backend-sdl/build.gradle
index 429076018..bcdc7f227 100644
--- a/backends/backend-sdl/build.gradle
+++ b/backends/backend-sdl/build.gradle
@@ -91,7 +91,7 @@ jnigen{
     add(Windows, x64){
         def path = "SDL2-$sdlVersion/x86_64-w64-mingw32"
         def root = "$rootDir/backends/backend-sdl/jni"
-        headerDirs += ["$path/include/SDL2"]
+        headerDirs += (String[])["$path/include/SDL2"]
         cppFlags += " " + execCmd("sh $root/$path/bin/sdl2-config --cflags")
         cFlags = cppFlags
         libraries = execCmd("sh $root/$path/bin/sdl2-config --static-libs") + " -lopengl32"
diff --git a/gradle/wrapper/gradle-wrapper.properties b/gradle/wrapper/gradle-wrapper.properties
index db9a6b825..0d1842103 100644
--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.10-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
diff --git a/extensions/freetype/build.gradle b/extensions/freetype/build.gradle
index d92cb25e9..043622978 100644
--- a/extensions/freetype/build.gradle
+++ b/extensions/freetype/build.gradle
@@ -10,7 +10,7 @@ jnigen{
     libsDir = file("libs").absolutePath

     all{
-        headerDirs += ["$fdir/include"]
+        headerDirs +=  (String[])["$fdir/include"]
         cIncludes = [
             //BASE
             "$fdir/src/base/ftsystem.c", "$fdir/src/base/ftinit.c", "$fdir/src/ba
diff --git a/backends/backend-sdl/build.gradle b/backends/backend-sdl/build.gradle
index bcdc7f227..9f12325b5 100644
--- a/backends/backend-sdl/build.gradle
+++ b/backends/backend-sdl/build.gradle
@@ -100,7 +100,7 @@ jnigen{
     add(Windows, x32){
         def path = "SDL2-$sdlVersion/i686-w64-mingw32"
         def root = "$rootDir/backends/backend-sdl/jni"
-        headerDirs += ["$path/include/SDL2"]
+        headerDirs +=  (String[])["$path/include/SDL2"]
         cppFlags += " " + execCmd("sh $root/$path/bin/sdl2-config --cflags")
         cFlags = cppFlags
         libraries = execCmd("sh $root/$path/bin/sdl2-config --static-libs") + " -lopengl32"
